### ISSUE TITLE
What about making history callback can record the information before training

### ISSUE THREAD

**Issue by MakeCent** [keras-team/keras#12846]
Current model.fit return a history which doesn't record the information (loss and metrics) before model train. Which means we cannot investigate the influence of the first epoch and the initialization method. I think it will be great if we can or can choice to let the history automatically record the loss and metrics of the model after initialization.
By the way, the epoch number store in history start with 0, which is not easy to read for end user. Defining epoch 0 as the one before training will also get benefit from this.
